From cb1ef3322671a67e2050a3eee18b49cdb4ed4bed Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Wed, 18 Jan 2023 20:54:41 +0100
Subject: [PATCH] libxdp: fix compilation on multiarch systems

Multiarch systems require an additional include path, which is covered
by ARCH_INCLUDES here. Just as lib/util, add it to BPF_CFLAGS.

Fixes compilation on debian:

In file included from xdp-dispatcher.c:3:
In file included from ../../headers/linux/bpf.h:11:
/usr/include/linux/types.h:5:10: fatal error: 'asm/types.h' file not found

Signed-off-by: Andre Heider <a.heider@gmail.com>
---
 lib/libxdp/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/libxdp/Makefile
+++ b/lib/libxdp/Makefile
@@ -28,7 +28,7 @@ PC_FILE := $(OBJDIR)/libxdp.pc
 TEMPLATED_SOURCES := xdp-dispatcher.c
 
 CFLAGS += -I$(HEADER_DIR)
-BPF_CFLAGS += -I$(HEADER_DIR)
+BPF_CFLAGS += -I$(HEADER_DIR) $(ARCH_INCLUDES)
 
 
 ifndef BUILD_STATIC_ONLY
